cowbird/handlers/impl/magpie.py

```diff
@@ -98,6 +122,69 @@ def get_resources_tree(self, resource_id):
             raise RuntimeError("Could not find the input resource's parent resources.")
         return resp.json()["resources"]
 
+    def get_children_resource_tree(self, resource_id):
+        # type: (int) -> List
+        """
+        Returns the associated Magpie Resource object and all its children.
+        """
+        resp = self._send_request(method="GET", url=f"{self.url}/resources/{resource_id}", params={"parent": "false"})
+        if resp.status_code != 200:
+            raise RuntimeError("Could not find the input resource's children resources.")
+        return resp.json()["resource"]["children"]
+
+    def get_geoserver_resource_id(self, workspace_name, layer_name, create_if_missing=True):
+        # type: (str, str, bool) -> int
+        """
+        Tries to get the resource id of a specific layer, on `geoserver` type services, and creates the resource and
+        workspace if they do not exist yet.
+        """
+        layer_res_id, workspace_res_id = None, None
+        geoserver_type_services = self.get_services_by_type("geoserver")
+        if not geoserver_type_services:
+            raise ValueError("No service of type `geoserver` found on Magpie while trying to get a layer resource id.")
+        for svc in geoserver_type_services.values():
+            if layer_res_id:
+                break
+            for workspace in self.get_children_resource_tree(svc["resource_id"]).values():
+                if workspace["resource_name"] == workspace_name:
+                    workspace_res_id = workspace["resource_id"]
+                    for layer in workspace["children"].values():
+                        if layer["resource_name"] == layer_name:
+                            layer_res_id = layer["resource_id"]
+                            break
+                    break
+        if not layer_res_id and create_if_missing:
+            if not workspace_res_id:
+                workspace_res_id = self.create_resource(
+                    resource_name=workspace_name,
+                    resource_type="workspace",
+                    parent_id=list(geoserver_type_services.values())[0]["resource_id"])
+            layer_res_id = self.create_resource(
+                resource_name=layer_name,
+                resource_type="layer",
+                parent_id=workspace_res_id)
+        return layer_res_id
+
+    def create_user(self, user_name, email, password, group_name):
+        resp = self._send_request(method="POST", url=f"{self.url}/users",
+                                  json={
+                                      "user_name": user_name,
+                                      "email": email,
+                                      "password": password,
+                                      "group_name": group_name
+                                  })
```

Comment:
True, this function is actually only used for testing. It was convenient while implementing to put it here with all the other functions that do API calls to Magpie, but it would make more sense to put them apart in the test files.

Reply:
That's the conclusion I came through after continuing the PR review. Move to extended test class is the right way to do it.